### ISSUE TITLE
DeviceElementConfiguration Offsets Needs Definition and Cleanup

### ISSUE THREAD

**Issue by knelson-farmbeltnorth** [ADAPT/Standard#32]
This is what exists today:
DeviceElementConfiguration.Offsets (conceivably intended for up to the 3 directional offsets on any type of configuration but relative to what?)
ImplementConfiguration.ControlPoint (What does  "control" mean here?   In the past I've interpreted this as the Implement Reference Point as relative to the Hitch, but this is redundant to defining the offsets on the Hitch)
ImplementConfiguration.YOffset (no idea what this is)
ImplementConfiguration.VerticalCuttingEdgeZOffset (is the cutting edge vertical, or is the "vertical" redundant to z?   Whatever the case, this property seems to be too specific of a definition within the universe of implement types)
ImplementConfiguration.GPSReceiverZOffset (perhaps intended for a supplemental GPS receiver on the implement, but then what of its X&Y offsets?)
SectionConfiguration.LateralOffset (I interpret this as the Y offset from the Implement Reference Point)
SectionConfiguration.InlineOffset (I interpret this as the X offset from the ImplementReferencePoint)
MachineConfiguration.GpsReceiverXOffset, GpsReceiverYOffset, GpsReceiverZOffset (conceivably relative to the tractor/vehicle reference point)

In general industry practice (following from ISO11783-10), there are defined reference points on each physical piece of equipment.  On a tractor/self-propelled implement, it is the center of the primary driving axle, on an implement it is similarly a center location at the (front) axle.    From the tractor reference point, there is a set of offsets to the GPS receiver.   There is then an additional set of offsets from the reference point for each hitch.   For the implement, there is a set of offsets from the reference point to its hitch.  There are then additional offsets from the implement reference point to each section, row, etc. that has its own physical definition.

The intent of what was originally implemented in ADAPT is unclear, potentially resulting from different authors at different times.  @strhea can you provide and history on the above and validate my interpretations?

Going forward I'm inclined to define an Offset entity that has X, Y and Z attributes and use that entity in these situations, with implementation guidance in comments:
MachineConfiguration.GPSReceiverOffset
HitchPoint.ReferencePointOffset
SectionConfiguration.Offset

The above requires that we clarify a "Section" in ADAPT as any device element at a lower level than the Implement, representing not only what ISO11783 calls a "section" (a controllable part of a boom), but also the ISO11783 "unit" (e.g., a nozzle or row unit with a defined geometry but not independently controllable).    Sections as children of other sections is possible today in ADAPT, we simply need to clarify what the Offset relates to.  From industry practice, this is always the Implement Reference point, and not the sub-sections parent section.

Additionally, we need to confirm that we follow ISO11783's definition of offset directions:  Forward=positive, Right=positive, and (non-intuitively) Down=Positive.

**Comment by knelson-farmbeltnorth** [comment]
Per discussion with @strhea, I understand that this part of the model was known to have issues/loose ends at the time of implementation.   As such, I made the following changes to make the model consistent with ISO-11783 and standard industry practice of modeling implement geometries:

1. Removed DeviceElementConfiguration "Offsets".   This property lacks definition of what the offsets are relative to (is the implement relative to itself?) and is redundant to other specific offset properties on the child types.
2. Changed  Hitch Point "Reference Point" to "Reference Point Offsets" for clarity.  It defines offsets relative to a reference point vs. defining a reference point itself.
3. Made a "GPS Receiver Offsets" instance of "Reference Point Offsets" replacing the 3 separate offsets on Machine Configuration and added to Implement Configuration for cases of a secondary GPS receiver on implements.
4. Merged ImplementConfig Physical Width & Width into a single Width property.   This was already discussed in a committee meeting as a redundancy.
5. Removed ImplementConfig "Control Point" - we can add this back if anyone can define it.
6. Removed ImplementConfig "YOffset" - see "Offsets" above:   YOffset to what?   An implement's reference location is the center of its axle and its offset to anything else is defined by the offsets on its hitches from this location.
7. Removed ImplementConfig "VerticalCuttingEdgeZOffset"   This had the feel of something added as a corner case that may be more appropriately handled in a context item or future enhancement to the model at such time that data may exist.
8. Standardized naming on SectionConfiguration
   a. SectionWidth -> Width
   b. InlineOffset -> X Offset
   c. LateralOffset -> Y Offset
